src/cpp/tests/testthat/run-tests.R

```diff
@@ -0,0 +1,18 @@
+library(testthat)
+
+runAllTests <- function(sourceDir, outputDir, filter = NA)
+{
+   testThatDir <- file.path(sourceDir, "tests/testthat")
+   
+   if (!is.na(filter))
+   {
+      tests <- testthat::test_dir(testThatDir, filter = filter)
+   } else
+   {
+      tests <- testthat::test_dir(testThatDir)
+   }
+   
+   resultFile <- file(file.path(outputDir, "testthat-failures.log"), open="w")
+   cat(sum(as.data.frame(tests)$failed), file = resultFile)
```

Comment:
nit: if you provide the file path as a string, R will take care of opening and closing the connection for you (ensuring it's closed if an error occurs, for example)